### PR TITLE
docs: cloudpickle is an interesting extension of the pickle functionality

### PR DESCRIPTION
**Pull request by hfwittmann**
## Description

Quote from their github: "cloudpickle makes it possible to serialize Python constructs not supported by the default pickle module from the Python standard library."

Its already implicity supported by kedro / kedro-plugins, but would be useful to have it in the docs.

## Development notes

Changed the documentation
Added cloudpickle to the unit tests

## Checklist

- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [x] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the relevant `RELEASE.md` file
- [x] Added tests to cover my changes

### CODE REVIEW

**Review by noklam — APPROVED** [review]
Thanks for the PR!

**Comment by datajoely** [comment]
Love this! Question for the group, should we think about changing the default pickle engine to something a bit more ML ready?

**Comment by marrrcin** [comment]
Cloudpickle is really good, we use it by default in Kedro-AzureML/Sagemaker/VertexAI 💪🏻

File: kedro-datasets/setup.py
Comment (marrrcin) [inline kedro-datasets/setup.py:155]: There's a whitespace at the end of this string.

**Review by marrrcin — CHANGES_REQUESTED** [review]

**Comment by stichbury** [comment]
> Its already implicity supported by kedro / kedro-plugins, but would be useful to have it in the docs.

Would it be possible to make an issue over on the [Kedro](https://github.com/kedro-org/kedro/issues) list please with the docs you think are needed?

**Comment by hfwittmann** [comment]
> > Its already implicity supported by kedro / kedro-plugins, but would be useful to have it in the docs.
> 
> Would it be possible to make an issue over on the [Kedro](https://github.com/kedro-org/kedro/issues) list please with the docs you think are needed?

Just to clarify: An issue or a pull-request (ie this same pull-request presumably)?

**Comment by stichbury** [comment]
Hi @hfwittmann 

Hope this answers your question although I may have misinterpreted it. You wouldn't be able to make docs changes on this PR because the docs are in a different repo, and we don't expect you to do that task too, although of course it's great if can contribute the changes.

Feel free to raise an issue over on https://github.com/kedro-org/kedro/issues to record the fact the work is done and then, if you do want to do that change, feel free to make a separate PR.

Jo

**Comment by hfwittmann** [comment]
> Hi @hfwittmann
> 
> Hope this answers your question although I may have misinterpreted it. You wouldn't be able to make docs changes on this PR because the docs are in a different repo, and we don't expect you to do that task too, although of course it's great if can contribute the changes.
> 
> Feel free to raise an issue over on https://github.com/kedro-org/kedro/issues to record the fact the work is done and then, if you do want to do that change, feel free to make a separate PR.
> 
> Jo

Excellent, I understand now.

Just to be clear, do I need to do this before the pull request is approved?

**Comment by stichbury** [comment]
> Just to be clear, do I need to do this before the pull request is approved?

Hi @hfwittmann No, there's no dependency between them. I have re-requested review from @marrrcin as someone well-placed to assess your PR, and all being well, with second approval, we can merge your change.

**Review by marrrcin — APPROVED** [review]

**Comment by stichbury** [comment]
@merelcht I didn't want to merge this without your signoff (it has 2 approvals but wanted to confirm it's OK to go in). Feel free to merge if you're happy!

**Comment by merelcht** [comment]
> @merelcht I didn't want to merge this without your signoff (it has 2 approvals but wanted to confirm it's OK to go in). Feel free to merge if you're happy!

I updated the release notes and now it's all ready to go! 👍

**Pull request merged**
